OneCommunityGlobal/HighestGoodNetworkApp
# Jianjun and RJ - Performance Optimization

### Description
Complete the priority medium task of LEADERBOARD COMPONENT
Improve loading speed of other people’s Time Log: When I click the time bar in the leaderboard for someone, the loading speed to see their time log is really slow. I’d like to speed that up if possible. 
Remaining major open problems proposed by RJ are described in the analysis [here](https://www.dropbox.com/sh/e43le4bts99dxuw/AADHoHLWQVaf1FDHdqRgU2gNa?dl=0) 
To sum up, major points to check:
1. avoid unnecessary re-render
2. check the dependency of the hooks
3. convert the Timelog component from class to function

### Mainly changes explained:
1. set the value of prop `isDashboard` as true when the timelog component is called at the main page, remove a duplicated condition.
2. Introduce the state `summaryBarData` and redesign the passing process of the data for the SummaryBar component so that it won’t use the ones from the leader board.
3. Convert the Timelog component from class to function and use the hooks and states to control the re-render.

### How to test:
To test if the component runs normally:
Log in --> see dashboard --> click timelog page ==> shows the correct output
log out and log in as another account --> see dashboard ===>shows the correct output
add time entry ===>shows the correct output
To test the performance:
add console.log in the code and check console output to see how many times each part/function runs when open the page.
